tendrl/gluster_integration/objects/volume/atoms/delete/__init__.py

```diff
@@ -88,8 +88,25 @@ def run(self):
             try:
                 # Acquire lock before deleting the volume from etcd
                 # We are blocking till we acquire the lock
+                # the lock will live for 60 sec after which it will be released.
+                locked = False
                 lock = etcd.Lock(NS.etcd_orm.client, 'volume')
-                lock.acquire(blocking=True,lock_ttl=None)
+
+                while not locked:
+                    try:
+                        # with ttl set, lock will be blocked only for 60 sec
+                        # after which it will raise lock_expired exception.
+                        # if this is raised, we have to retry for lock
+                        lock.acquire(blocking=True,lock_ttl=60)
```

Comment:
whats the reasoning behind choosing a 60 second ttl? high ttls will increase idling of the sync loop and any other volume operation

The lock should live approximately as long as the lock is required. In case of delete, a 10 second TTL would suffice